src/components/lend/LendAnvil.tsx

```diff
@@ -56,13 +56,13 @@ export const LendAnvil: React.FC<ReserveComponentProps> = ({ poolId, assetId })
   const freeUserBalanceScaled = Number(userBalance) / scalar - stellar_reserve_amount;
 
   const handleLendAmountChange = (lendInput: string) => {
-    let regex = new RegExp(`^[0-9]*\.?[0-9]{0,${decimals}}$`);
-    if (regex.test(lendInput) && userPoolData && reserve) {
+    setToLend(lendInput);
+    if (userPoolData && reserve) {
       let num_lend = Number(lendInput);
       let lend_base = num_lend * assetPrice * reserve.getCollateralFactor();
       let tempEffectiveCollateral = userPoolData.estimates.totalEffectiveCollateral + lend_base;
+      /**  @dev @TODO  how should this number behave in UI */
```

Comment:
which number? the newEffectiveCollateral?

Reply:
yes , the values that are only calculated in that moment when the value changes but the conditions for calculating it are not met for example